kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/api/common/LocalRef.java

```diff
@@ -9,21 +9,26 @@
 import java.util.Comparator;
 import java.util.Objects;
 
+import edu.umd.cs.findbugs.annotations.Nullable;
+
 /**
  * Abstraction for references in one kubernetes resource to some kubernetes resource in the same namespace.
  * Two LocalRefs are equal iff they have the same group, kind and name (they don't need to have the same class)
  * @param <T> The Java type of the resource
  */
 public abstract class LocalRef<T> implements Comparable<LocalRef<T>> {
 
-    public static final Comparator<LocalRef<?>> COMPARATOR = Comparator.<LocalRef<?>, String> comparing(LocalRef::getKind)
-            .thenComparing(LocalRef::getGroup)
-            .thenComparing(LocalRef::getName);
+    public static final Comparator<LocalRef<?>> COMPARATOR = Comparator.<LocalRef<?>, String> comparing(LocalRef::getKind, Comparator.nullsLast(String::compareTo))
```

Comment:
what's the motivation here?  Null safety? I get that group might be null (core kube api), but I thought that kind and name were mandatory.